docs/source/developer/architecture/5_building_block_view.rst

```diff
@@ -7,10 +7,64 @@
 5. Building Block View
 ======================
 
+The layers of fotoobo
+---------------------
+
+There are three designated layers in **fotoobo**. These are the interface layer, the business logic
+layer and the infrastructure layer.
+
+.. image:: diagrams/fotoobo_layers.drawio.svg
+  :width: 100%
+  :alt: The layers of fotoobo visualized
+
+In this diagram you see the designated paths to use when accessing a Fortinet device. Although these 
+are the recommended ways to use it, you may also access the layers in other ways. Just keep in mind 
+there always should be a rational and understandable approach.
+
+
+The Interface Layer
+^^^^^^^^^^^^^^^^^^^
+
+This is the main entry point for interaction with **fotoobo** when you installed it as an
+application on your system. It acts as a frontend for users and automated tools. At the Moment only
+the CLI part is implemented. In future version an API may be available if there is a need for it.
+
+Call **fotoobo** with its command(s) directly from the command line or from any automation engine
+like cron jobs, Rundeck or others.
+
+The Business Logic Layer
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+At this level you have access to specific use cases. These are called **tools** in **fotoobo**. Any
+CLI command or API endpoint should point to such a **tool** which then interacts with the
+infrastructure layer.
+
+Whenever you use **fotoobo** as a module in your own code (instead of using it as an installed
+application) you may directly access this layer.
+
+The Infrastructure Layer
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this lowest level of **fotoobo** we directly interact with the infrastructure, meaning we handle 
+the authentication and API calls to the devices like FortiGate, FortiManger and others. The 
+interface layer should not directly access this infrastructure layer. Instead there should always 
+be a **tool** at the business logic layer which connects these two. No other layer than the 
```

Comment:
I think it should be "a tool **in** the business logic layer, because "at" usually points to a physical location.

Reply:
fixed